src/public/javascripts/ui.js

```diff
@@ -1,282 +1,268 @@
-  });
-}
-
-/**
- * Fills constants in settings modal
- */
-function fillConstants() { // eslint-disable-line no-unused-vars
-  CONFIG.updateConstants();
-  document.getElementById('formFeedback').innerHTML = 'Will restart for changes to take place.';
-  document.getElementById('podIP').value = String(CONFIG_CONSTANTS.serverAddr.ip);
-  document.getElementById('podPort').value = CONFIG_CONSTANTS.serverAddr.port;
-  document.getElementById('scanningRate').value = CONFIG_CONSTANTS.dataSendRate;
-  document.getElementById('lvBoneIP').value = CONFIG_CONSTANTS.lvBone.ip;
-  document.getElementById('lvBonePort').value = CONFIG_CONSTANTS.lvBone.port;
-  document.getElementById('hvBoneIP').value = CONFIG_CONSTANTS.hvBone.ip;
-  document.getElementById('hvBonePort').value = CONFIG_CONSTANTS.hvBone.port;
-  document.getElementById('renderInterval').value = CONFIG_CONSTANTS.renderInterval;
-}
-
-// Window Handling
-
-document.getElementById('min-window').addEventListener('click', () => {
-  ELECTRON_WINDOW.minimize();
-});
-
-document.getElementById('max-window').addEventListener('click', () => {
-  if (!ELECTRON_WINDOW.isMaximized()) ELECTRON_WINDOW.maximize();
-  else ELECTRON_WINDOW.unmaximize();
-});
-
-document.getElementById('close-window').addEventListener('click', () => {
-  ELECTRON_WINDOW.close();
-});
+/**
+ * @module UI
+ * @author Eric Udlis, Luke Houge
+ * @description Handles all responsive UI elements of the dashboard
+ */
+
+const { remote: REMOTE } = require('electron');
+
+const ELECTRON_WINDOW = REMOTE.getCurrentWindow();
+const CONFIG = require('./public/javascripts/config');
+const CONFIG_CONSTANTS = require('./public/javascripts/config').constants;
+
+const RATE = CONFIG_CONSTANTS.DATA_SEND_RATE;
+
+/*
+Modals
+Purpose: code for opening a pop up modal box
+*/
+const SETTINGS_MODAL = document.querySelector('.settingsModal');
+const SETTINGS_TRIGGER = document.getElementById('settingsTrigger');
+const CLOSE_BUTTON = document.querySelector('.close-button');
+
+/**
+ * Toggles hide/show of the settings modal
+ */
+function toggleSettingsModal() {
+  SETTINGS_MODAL.classList.toggle('show-modal');
+  fillConstants(); // eslint-disable-line no-use-before-define
+}
+
+if (SETTINGS_TRIGGER) SETTINGS_TRIGGER.addEventListener('click', toggleSettingsModal);
+if (CLOSE_BUTTON) CLOSE_BUTTON.addEventListener('click', toggleSettingsModal);
+
+// window.addEventListener('click', windowOnClick);
+
+/*
+Focus Clone
+Purpose: Fill boxes at the top with live information from tables, for better visibility
+*/
+// Counters for Focus Header
+let focusOne;
+let focusTwo;
+let focusThree;
+let focusFour;
+
+const settingsSubmit = document.getElementById('podSettingsSubmit');
+
+
+// filling for focus clone
+let x = 1; // counter for boxes filed so far
+/**
+ * Clones the value from a table to the focus header
+ * @param {String} id ID of sensor to focus
+ */
+function clone(id) { // eslint-disable-line no-unused-vars
+  if (x === 1) {

```

Comment:
Very repeated, just have an if statement checking if x is in the bounds you expect, and if it is you can replace the hardcoded stuff below by just using x as the number in there. If this comment is too confusing reach out to me on Slack

Reply:
@LukeHouge I tried to do this but again, don't understand enough about what this does to fix it.

Reply:
Hmm I'm not sure how this could be condensed. It relies on keeping each individual function active to keep updating them, so if it was condensed to a single if statement when a new one is activated the old one would be deactivated. I might be missing something, please let me know if I am.

Reply:
@LukeHouge Also make sure your tests pass before pushing your commits

Reply:
I dont understand why it would overwrite a previous interval? Even something like this should work afaik:

```javascript
if (x >= 0 && x <= 4) {
focus[x] = setInterval(() => {
      const value = Number(document.getElementById(id).innerHTML);
      document.getElementById('header_value_' + i).innerHTML = value;
      let name = id.replace(/([a-z\xE0-\xFF])([A-Z\xC0\xDF])/g, '$1 $2'); // changes the ID from camel case to regular
      name = name.charAt(0).toUpperCase() + name.slice(1); // Capitalizes first letter
      document.getElementById('header_label_' + i).innerHTML = name;
}, RATE);
x += 1;  // x is still a dirty name for a global variable
}
```

Reply:
@LukeHouge @EUdds Then every time clone is called, it sets a different interval, should leave the previous one intact

Reply:
@LukeHouge

Reply:
@LukeHouge

Reply:
Yeah I tried this and it adds the new one but the past ones get stopped and no longer update. Im not sure why it seems to close the old interval when a new one is created.

Reply:
I researched it a bit and it is something about js being single-threaded. Not sure why keeping them in if statements gets around that though

Reply:
It was a race condition with that global variable and the interval I think